include/core/shape.h

```diff
@@ -531,4 +555,217 @@ class CompositeShape : public Shape<dim>
   std::vector<std::shared_ptr<Shape<dim>>> components;
 };
 
+
+// RBF Shapes express the signed distance function as a linear combination of
+// Radial Basis Functions, which have a defined support radius and basis
+// function. A collection of nodes and weights compose the object.
+template <int dim>
+class RBFShape : public Shape<dim>
+{
+public:
+  /**
+   * Class taken from Optimad Bitpit. https://github.com/optimad/bitpit
+   * @enum RBFBasisFunction
+   * @ingroup RBF
+   * @brief Enum class defining types of RBF kernel functions that could be used in bitpit::RBF class
+   */
+  enum class RBFBasisFunction : int
```

Comment:
Yes it needs to be an int, because the basis function type is written as a number when it comes out of bitpit. If we changed the data type it wouldn't be possible to load the data using fill_vectors_from_file().